### ISSUE TITLE
Upstream Statistics supports to be collected  per upstream_host

### ISSUE THREAD

**Issue by wenlxie** [envoyproxy/envoy#21177]
*Title*: *Upstream Statistics supports to be collected  per upstream_host*

*Description*:

There are many statistics for upstream clusters
https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#statistics
This statistics are collected per cluster, if there are just few of the upstream host has issues like: 
More 5xx responses, More higher latency, More connection failed, it is not easy to find which upstream host has issues based on the statistics.  
If these statistics can be collected per upstream host, then it can be more useful when debug issues.

**Comment by lizan** [comment]
cc @jmarantz

**Comment by mattklein123** [comment]
I think this would be an interesting feature. Obviously it must be opt-in for cardinality reasons.

**Comment by zhangbo1882** [comment]
It must be helpful if envoy can provide such statistics.

**Comment by jmarantz** [comment]
Would it be helpful to be able to enable such statistics with an admin POST when the need arises, maybe just the hosts of a specific cluster?

**Comment by wenlxie** [comment]
> Would it be helpful to be able to enable such statistics with an admin POST when the need arises, maybe just the hosts of a specific cluster?

Yes, by default it is not enabled, and this can be enabled by an admin POST or by cluster configuration.

**Comment by ramaraochavali** [comment]
There are some level of stats in /clusters output per host I think?

**Comment by wenlxie** [comment]
> There are some level of stats in /clusters output per host I think?

@ramaraochavali  
Thanks for reply, 
There are just few statistics in /clusters output there

```
outbound|8082||a.b.c.com::10.100.200.100:8082::cx_active::43
outbound|8082||a.b.c.com::10.100.200.100:8082::cx_connect_fail::45
outbound|8082||a.b.c.com::10.100.200.100:8082::cx_total::308343
outbound|8082||a.b.c.com::10.100.200.100:8082::rq_active::2
outbound|8082||a.b.c.com::10.100.200.100:8082::rq_error::47
outbound|8082||a.b.c.com::10.100.200.100:8082::rq_success::86474602
outbound|8082||a.b.c.com::10.100.200.100:8082::rq_timeout::0
outbound|8082||a.b.c.com::10.100.200.100:8082::rq_total::86474818
outbound|8082||a.b.c.com::10.100.200.100:8082::hostname::
outbound|8082||a.b.c.com::10.100.200.100:8082::health_flags::healthy
outbound|8082||a.b.c.com::10.100.200.100:8082::weight::1
outbound|8082||a.b.c.com::10.100.200.100:8082::region::lvs
outbound|8082||a.b.c.com::10.100.200.100:8082::zone::
outbound|8082||a.b.c.com::10.100.200.100:8082::sub_zone::
outbound|8082||a.b.c.com::10.100.200.100:8082::canary::false
outbound|8082||a.b.c.com::10.100.200.100:8082::priority::0
outbound|8082||a.b.c.com::10.100.200.100:8082::success_rate::-1.0
outbound|8082||a.b.c.com::10.100.200.100:8082::local_origin_success_rate::-1.0
```
Compare to the statistics for cluster https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#statistics, this is really not enough.

**Comment by moonlightysh** [comment]
This is nice catch.
